test/compilable/b22082.d

```diff
@@ -0,0 +1,48 @@
+// PERMUTE_ARGS:
+// REQUIRED_ARGS: -betterC
+
+struct VertexA
+{
+    float pos;
```

Comment:
All floating point types should be tested, not only `float`.

Reply:
i tried to update the test but i'm getting the following:

```
test.obj : error LNK2019: unresolved external symbol _memset80 referenced in function _D4test__TQiTeZQnFNaNbNiNfZv
test.exe : fatal error LNK1120: 1 unresolved externals
Error: linker exited with status 1120
```

updated test:

```D
// PERMUTE_ARGS:
// REQUIRED_ARGS: -betterC

struct VertexA(T)
{
    T pos;
}

struct VertexB(T)
{
    T[2] pos;
}

void test(T)()
{
    T f_a;
    T f_b;
    assert(f_a != f_b);

    f_a = 0;
    assert(f_a != f_b);

    f_b = 0;
    assert(f_a == f_b);


    VertexA!T a_a;
    VertexA!T a_b;
    assert(a_a != a_b);

    a_a.pos = 0;
    assert(a_a != a_b);

    a_b.pos = 0;
    assert(a_a == a_b);

    VertexB!T b_a;
    VertexB!T b_b;
    assert(b_a != b_b);

    b_a.pos = 0;
    assert(b_a != b_b);

    b_b.pos = 0;   
    assert(b_a == b_b);
}

extern(C) int main()
{
    test!(float)();
    test!(double)();
    test!(real)();

    return 0;
}
```